### ISSUE TITLE
question about loss function

### ISSUE THREAD

**Issue by zhaoxingwudy** [taeoh-kim/Pytorch_InfoGAN#1]
# in infogan.py, what's the meaning of   
 ```python
 d_loss_dc = -(torch.mean(torch.sum(dc * output_dc, 1)) + torch.mean(torch.sum(dc * dc, 1)))
```    
why do not use cross entropy loss ? why not use the log in the first item? and the second item is a constant?

**Comment by yixuanren** [comment]
As I guess, there might be a typo, and the correct version should be:

`d_loss_dc = -(torch.mean(torch.sum(dc * output_dc, 1)) + torch.mean(torch.sum(dc * dc, 1)))`

Then it looks like the difference between the dot product of the groud truth and the actual output, and the dot product when output is perfect.
